pkg/util/proxy/proxy.go

```diff
@@ -94,14 +102,39 @@ func NewThrottledUpgradeAwareProxyHandler(location *url.URL, transport http.Roun
 	return proxy.NewUpgradeAwareHandler(location, transport, wrapTransport, upgradeRequired, proxy.NewErrorResponder(responder))
 }
 
+func GetTlsConfigForCluster(ctx context.Context, cluster *clusterapis.Cluster, secretGetter SecretGetterFunc) (*tls.Config, error) {
+	if cluster.Spec.SecretRef == nil {
```

Comment:
Can you remind me when the secret would not exist?

Reply:
From my point of view, it couldn't be empty. But I saw it's an optional field in the struct comment.

Reply:
For `Pull` mode cluster, the `karmada-agent` may not set the secret, see the usage from `karmada-agent`:
```bash
      --report-secrets strings                                                                                                                                                                     
                The secrets that are allowed to be reported to the Karmada control plane during registering. Valid values are 'KubeCredentials', 'KubeImpersonator' and 'None'. e.g
                'KubeCredentials,KubeImpersonator' or 'None'. (default [KubeCredentials,KubeImpersonator])

```

Reply:
```suggestion
	// The secret is optional for a pull-mode cluster, if no secret just returns a config with root CA unset.
	if cluster.Spec.SecretRef == nil {
```
Maybe we can have a comment here.